### PR TITLE
Implement Kafka Streams exception handlers

### PR DESCRIPTION
**Pull request by nscuro**
### Description

This PR adds custom handlers for deserialization, processing, and production exceptions in Kafka Streams.

### Addressed Issue

Partially addresses https://github.com/DependencyTrack/hyades/issues/674

### Additional Details

See https://github.com/DependencyTrack/hyades/issues/674 for details.

### Checklist

- [x] I have read and understand the [contributing guidelines](../CONTRIBUTING.md#pull-requests)
- ~This PR fixes a defect, and I have provided tests to verify that the fix is effective~
- [x] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- ~This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)~
- ~This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly~

### CODE REVIEW

File: src/main/java/org/dependencytrack/event/kafka/exception/KafkaStreamsUncaughtExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public StreamThreadExceptionResponse handle(final Throwable exception) {
+        final Throwable rootCause = ExceptionUtils.getRootCause(exception);
+
+        if (rootCause instanceof TimeoutException
+                || rootCause instanceof ConnectTimeoutException
```
Comment (VithikaS) [inline src/main/java/org/dependencytrack/event/kafka/exception/KafkaStreamsUncaughtExceptionHandler.java:57]: Should we add ReadTimeoutException also ?

**Review by VithikaS — DISMISSED** [review]
Looks good, just added a minor question in PR.

File: src/main/java/org/dependencytrack/event/kafka/exception/KafkaStreamsUncaughtExceptionHandler.java (hunk 1 above)
Comment (nscuro) [reply src/main/java/org/dependencytrack/event/kafka/exception/KafkaStreamsUncaughtExceptionHandler.java:57]: Good point, will add it!

Comment (nscuro) [reply src/main/java/org/dependencytrack/event/kafka/exception/KafkaStreamsUncaughtExceptionHandler.java:57]: Hmmm, there exists no `ReadTimeoutException`... But I do also recall seeing this class before, but perhaps in another code base.

**Review by VithikaS — APPROVED** [review]

**Pull request merged**